src/workerd/tools/api-encoder.c++

```diff
@@ -28,8 +34,8 @@
 #include <workerd/api/trace.h>
 #include <workerd/api/urlpattern.h>
 #include <workerd/api/node/node.h>
-#include <workerd/io/compatibility-date.h>
-#include <workerd/jsg/rtti.h>
+
+#if !API_ENCODER_HDRS_ONLY
```

Comment:
With such a large `#if..#endif` block, we could consider splitting this file into two files, something like "tools/workerd-apis.h" and "api-encoder.c++".

Reply:
I'm kind of torn on this but have a slight preference for keeping it as-is. Having a separate file for the headers would make the AST generation input more clean, but it would also mean having a file just containing 19 include directives that is of little use elsewhere (it could be used to include API headers elsewhere, but in my include cleanup patches I have been trying to avoid including the entire API when this is not strictly required; the size of the includes tends to negatively affect compile times). Overall, it makes sense to me referring to the api encoder source as it is also used for type generation and necessarily includes the entire API.